Sources/DbPackages-mysql.php

```diff
@@ -197,9 +197,12 @@ function smf_db_create_table($table_name, $columns, $indexes = array(), $paramet
 	}
 
 	$table_query .= ') ENGINE=' . $parameters['engine'];
-	if (!empty($db_character_set) && $db_character_set == 'utf8')
+	if (!empty($db_character_set) && $db_character_set == 'utf8' && !$smcFunc['db_mb4'])
```

Comment:
Optional, might be good move the db_mb4 check into a ternary inside the if to avoid having to double check for $db_character_set

```suggestion
	if (!empty($db_character_set) && $db_character_set == 'utf8')
	{
	    $table_query .=  ' DEFAULT CHARSET=utf8 . (!$smcFunc['db_mb4'] ? '  COLLATE=utf8_general_ci' : 'mb4 COLLATE=utf8mb4_general_ci');
	}
```

Or make it cleaner by defining the charset as a variable

Reply:
both in my eyes make the code harder to read, reason why i choose this way.

Reply:
Same ternary logic was applied on another PR: https://github.com/SimpleMachines/SMF2.1/pull/6273/files

Reply:
Your suggest code changes got more line of changes and
is harder to read as the existing one.

So thanks but no.

Reply:
I think having repetitive code is actually harder to read and can be confusing since both ifs are the same except for the last part but as you wish